akka-stream/src/main/scala/akka/stream/impl/UnfoldResourceSourceAsync.scala

```diff
@@ -102,7 +110,14 @@ import scala.util.control.NonFatal
     }
 
     private def createResource(): Unit = {
-      create().onComplete(createdCallback)
+      create().onComplete { resource ⇒
+        if (stoppedEarly) {
```

Comment:
Wouldn't it be safer to move this logic inside `createdCallback`?

Right now couldn't there still be an interleaving where `stoppedEarly` is read here, then `postStop` is executed, and only then `createdCallback` is invoked?

Reply:
No, because that will not be executed if the stage already stopped.